syntax = "proto3";
// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

option java_multiple_files = true;
option java_package = "io.grpc.examples.fastapi";
option java_outer_classname = "GrpcApiProto";
option objc_class_prefix = "RTG";

package fastapi;

service Saver {
  rpc SaveDoc(DocObj) returns (DocObj) {}
}

message DocObj {
  int64 article_id = 1;
  string text = 2;
  string title = 3;
  string date = 4;
  Lang lang = 5;
  message Location {
    float lat = 1;
    float lon = 2;
  }
  repeated Location locations = 6;
  message Vector {
    float value = 1;
  }
  message Keyword {
    string name = 1;
  }
  message Entity {
    string name = 1;
  }
  message Theme {
    string name = 1;
  }
  message Class {
    string name = 1;
  }
  repeated Vector semantic_vector = 7;
  repeated Keyword keywords = 8;
  repeated Entity entities = 9;
  repeated Theme themes = 10;
  repeated Class _class = 11;
}

enum Lang {
  RU = 0;
  EN = 1;
}
